src/python/impactx/dashboard/Input/latticeConfiguration/variable_handler.py

```diff
@@ -0,0 +1,270 @@
+    generalFunctions,
+)
+
+server, state, ctrl = setup_server()
+
+
+init_value = ""
+state.variables = [
+    {"name": init_value, "value": init_value, "error_message": init_value}
+]
+state.is_only_variable = len(state.variables) == 1
+
+
+class LatticeVariableHandler:
+    """
+    Stores all functionality for dashboard variable referencing.
+    """
+
+    @staticmethod
+    @state.change("variables")
+    def on_variables_list_change(variables, **kwargs):
+        for lattice_element in state.lattice_params_bound_or_pending_variable.values():
+            ctrl.updateLatticeElementParameters(
+                lattice_element["index"],
+                lattice_element["parameter_name"],
+                lattice_element["ui_value"],
+                lattice_element["parameter_type"],
+            )
+
+    # -----------------------------------------------------------------------------
+    # Controllers
+    # -----------------------------------------------------------------------------
+
+    @staticmethod
+    @ctrl.add("add_variable")
+    def on_add_change() -> None:
+        """
+        Adds a new variable to the dashboard's variable
+        with empty values and updates UI.
+        Stored in a state which contains a list with dictionaries.
+        """
+
+        new_variable = {key: "" for key in state.variables[0]}
+        state.variables.append(new_variable)
+        state.dirty("variables")
+        LatticeVariableHandler.update_delete_availability()
+
+    @staticmethod
+    @ctrl.add("delete_variable")
+    def on_delete_change(index: int) -> None:
+        """
+        Deletes the variable defined by the user
+        provided the index
+
+        :param index: The index of the variable
+        """
+
+        state.variables.pop(index)
+        state.dirty("variables")
+        LatticeVariableHandler.update_delete_availability()
+
+    @staticmethod
+    @ctrl.add("update_variable")
+    def on_variable_change(key_name: str, index: int, event) -> None:
+        """
+        Called when any variable name or value changes.
+        Validates the new value and updates it's associated sim value.
+
+        :param key_name: The name of the variable.
+        :param index: The index of the variable.
+        :param event: Either the variable's new name or value.
+        """
+
+        if key_name == "name":
+            LatticeVariableHandler.validate_variable_name(event, index)
+            state.variables[index]["name"] = event
+        else:
+            state.variables[index][key_name] = event
+        state.dirty("variables")
+
+    @staticmethod
+    @ctrl.add("reset_variables")
+    def on_reset_variables() -> None:
+        """
+        Resets the dashboard's variables to default.
+        """
+
+        state.variables = [
+            {"name": init_value, "value": init_value, "error_message": init_value}
+        ]
+        state.dirty("variables")
+        LatticeVariableHandler.update_delete_availability()

```

Comment:
Could this manual call be removed if `update_delete_availability` is decorated with `@state.change("variables")`?
```suggestion
```